### ISSUE TITLE
About code for evaluation on CAPE dataset

### ISSUE THREAD

**Issue by boqian-li** [riccardomarin/NICP#9]
Hi @riccardomarin , I am trying to test NICP on dataset CAPE. I have got pretrained model as training branch readme shows but I didn't find the evaluation code for CAPE dataset. Could you please provide more info about that? Thanks!

**Comment by boqian-li** [comment]
@riccardomarin Hi, I successfully evaluated on CAPE dataset by modifying evaluation_benchmark.py. But it needs the arg: core.guess_rot=True.

**Comment by riccardomarin** [comment]
Hi @boqian-li ,
Thanks for taking the initiative. I was going to suggest a similar approach: evaluating CAPE requires similar pipeline described here for FAUST: running the "evaluation_benchmark.py" on the CAPE test set after preparing the data similar to PTF, and then computing the error w.r.t. ground-truth registration.

However, the guess_rot parameter is not needed, and it was never used in the paper's experiments. If I remember correctly, canceling the rotation inside the evaluation script (i.e., set alpha=0) should be sufficient to obtain correctly aligned shapes (although now I am curios how well it performs with guess_rot=True). 

In the while, I am working on a script for CAPE, but it might take some time.

**Comment by boqian-li** [comment]
Hi @riccardomarin , luckily it performs perfectly as it should. 
You said "preparing the data similar to PTF" before evaluating on CAPE test set, but I directly test each **clothed mesh obj** in CAPE test set with "demo mode", which also performs perfectly. 

Specifically, I just ran this command `PYTHONPATH=. python ./src/lvd_templ/evaluation/evaluation_benchmark.py core.guess_rot=True` and manually change the data path to the folder of CAPE clothed mesh objs ( I extracted them from the vanilla CAPE data)

Could you please tell me if it's the same process as what is in the paper experiments?

**Comment by riccardomarin** [comment]
By "preparing the data" I meant (in the case your goal is to fully replicate the quantitative evaluation) taking the same frames and test subjects provided in PTF code. 

Apart from the `guess_rot` (that I didn't use), it should be the same process as the paper experiments.

Glad it worked well!

**Comment by boqian-li** [comment]
Hi @riccardomarin , I tried to reproduce the experiment on CAPE dataset. 
I used the pretrained model and evaluate it on 1021 scans which are from the same frames and test subjects as those described in the paper. And I used the `evaluation_benchmark.py` with CAPE scan folder(extracted from vanilla CAPE dataset by using the official utils). After completing the prediction process, I evaluated the result by computing mean v2v distance between pred smpls (with or without chamfer refinement) and ground-truth smpls (extracted from vanilla CAPE dataset by using the minimal_body_mesh as template_v and 10pc SMPL model pkl with corresponding gender). 
But I got different result from that in the paper: 1.89 (without chamfer refinement) 1.53 （with chamfer refinement) 
Could you please tell me if there's any difference from the original experiment?

**Comment by boqian-li** [comment]
> Hi @riccardomarin , I tried to reproduce the experiment on CAPE dataset. I used the pretrained model and evaluate it on 1021 scans which are from the same frames and test subjects as those described in the paper. And I used the `evaluation_benchmark.py` with CAPE scan folder(extracted from vanilla CAPE dataset by using the official utils). After completing the prediction process, I evaluated the result by computing mean v2v distance between pred smpls (with or without chamfer refinement) and ground-truth smpls (extracted from vanilla CAPE dataset by using the minimal_body_mesh as template_v and 10pc SMPL model pkl with corresponding gender). But I got different result from that in the paper: 1.89 (without chamfer refinement) 1.53 （with chamfer refinement) Could you please tell me if there's any difference from the original experiment?

actually, I suppose it's the difference between the ways we extract the ground truth smpl mesh because the ground truth scan vertices are just in the vanilla CAPE dataset with no chance to differ.

**Comment by boqian-li** [comment]
> Hi @riccardomarin , I tried to reproduce the experiment on CAPE dataset. I used the pretrained model and evaluate it on 1021 scans which are from the same frames and test subjects as those described in the paper. And I used the `evaluation_benchmark.py` with CAPE scan folder(extracted from vanilla CAPE dataset by using the official utils). After completing the prediction process, I evaluated the result by computing mean v2v distance between pred smpls (with or without chamfer refinement) and ground-truth smpls (extracted from vanilla CAPE dataset by using the minimal_body_mesh as template_v and 10pc SMPL model pkl with corresponding gender). But I got different result from that in the paper: 1.89 (without chamfer refinement) 1.53 （with chamfer refinement) Could you please tell me if there's any difference from the original experiment?

I wonder how you obtained the test subset with subsample-ratio==20?  with ratio==5 firstly and then ratio==4? And how you obtained the scan mesh with gt smpl mesh from CAPE dataset?

**Comment by boqian-li** [comment]
@riccardomarin Hi, I tried again with the scan mesh extracted using PTF code (cape.py). This time, I got a very close number: 1.77. Do you have any advice on how to improve the result to make it closer to the one in the paper? If there are any differences from your original experiment, please let me know. Thanks!

**Comment by Mecel1147** [comment]
Hi @riccardomarin, same question about SHREC19 dataset. Thanks!
